### ISSUE TITLE
Really remove jar?

### ISSUE THREAD

**Issue by beiriannydd** [oktadeveloper/okta-aws-cli-assume-role#127]
Purging the oktaawscli.jar and the joda-time jar from the history would drastically reduce the overhead for this repo.  It would mean history rewrites though, so I can understand if it's not a high priority.
```
100644 blob 4d54b2628f029db84e7965e8175102f65c4df87e  627814	lib/joda-time-2.9.3.jar
100644 blob 6a696deba8d69a56cd1e5c6234c6b30d5be558e9 3335850	out/oktaawscli.jar
100644 blob 3828bb98150f6e998b5f96a1730049339442d8b3 3336367	out/oktaawscli.jar
100644 blob 500653827cb3d0b33b914ac2461dac14b2a1d0ca 3336491	out/oktaawscli.jar
100644 blob a13b82da58409fe69dd64b94f761985184b6af89 4739469	out/oktaawscli.jar
100644 blob 73fec5692894dd74500314dc2a83a67f20289144 4739481	out/oktaawscli.jar
100644 blob a2c992e09210e51afe5edd6308a3f02dbcc0908a 4739492	out/oktaawscli.jar
100644 blob b1911c2f166795ad695c7250f9fcdcc9bf5cb512 4739963	out/oktaawscli.jar
100644 blob 1c2d054f90c7100bc9c8d879ffcf1e7be81f11b9 4740310	out/oktaawscli.jar
100644 blob 5d11fe5b60c033ee43445b0a42131125d191ef53 4740660	out/oktaawscli.jar
100644 blob d3290ddc68036a8176cbd6085719421f10bd34bf 4740667	out/oktaawscli.jar
100644 blob 7dd2ce5d763595b768dbe643eca5e9810d8c43b2 4740824	out/oktaawscli.jar
100644 blob 80b62d0ab1c93836156249e2cb91a71b278e2b5c 4740830	out/oktaawscli.jar
100644 blob 3de9eeb4dc0292e77182cf0cbe4b9bcca42064e1 4740831	out/oktaawscli.jar
100644 blob 8cc5fb5425974b2f9e0e8a386cc83c63a451c549 4740879	out/oktaawscli.jar
100644 blob d22779d05d761ad752431aaf2a9b29710c74b477 77404990	out/oktaawscli.jar
100644 blob e9ff7268abd6dac8cea2f52d8fa016a2344ea9f3 90088707	out/oktaawscli.jar
100644 blob 5ec542e18a626cc5b061d0ce702d2eb50ae76b5c 99746130	out/oktaawscli.jar
100644 blob d7277ba6b876f8d364af4700d32d3358b9aff247 99746157	out/oktaawscli.jar
100644 blob 273e8c449e7f921b92d5172b010adc56f6807b11 99746451	out/oktaawscli.jar
100644 blob 5d2e3d28ea04bfe42900636eca0ce93245117b5a 99747014	out/oktaawscli.jar
100644 blob e1d0b84725649ca33487ec66afb3fd7ce2d25717 99748201	out/oktaawscli.jar
100644 blob 61f84e7c2f5d18b48e97aa62b2fa1ad032c0a4c0 99748528	out/oktaawscli.jar
100644 blob abacf439800d1a0cabb94ed471ab0505f93f6570 99752483	out/oktaawscli.jar
100644 blob fd47310546aef7329d50d448f6ad6bee032d80a7 99754647	out/oktaawscli.jar
```
These all still live in the bare repo.

**Comment by beiriannydd** [comment]
`https://github.com/nachoparker/git-forget-blob/blob/master/git-forget-blob.sh` is a handy tool for cleaning single files.  It'd need a little tweaking for all these.

**Comment by beiriannydd** [comment]
Hi Mraible, this isn't the sort of thing that you can pull request, it modifies the history.  Anything merged via pull request can only increase history.

**Comment by AlainODea** [comment]
This is a neat idea. I support this. The historical JARs do dramatically bloat this repo a **lot**.

@mraible this is worth doing and is fairly straight-forward since we only have a master branch.

For a while afterwards, I recommend using only squash and merge on PRs to avoid reintroducing the bloated history. All PRs in flight at the time would need to be rebased (the contributors would need to do that).

The safest way to achieve this is for a maintainer with write access to do it themselves. A contributor could do it, but the validation effort would eclipse the effort of just doing it.

This is essentially the entire workflow for someone with write access to master:

```bash
git clone ssh://git@github.com/oktadeveloper/okta-aws-cli-assume-role.git
cd okta-aws-cli-assume-role/
git forget-blob out/oktaawscli.jar
git forget-blob lib/joda-time-2.9.3.jar
git forget-blob lib/okta-sdk.jar
# seems like git-forget-blob drop the remotes
git remote add origin ssh://git@github.com/oktadeveloper/okta-aws-cli-assume-role.git
git push --force origin master
```

I've done this locally and pushed it to https://github.com/AlainODea/okta-aws-cli-assume-role/tree/master. The repo shrinks fro 293.48 MiB to 900KiB.

Reference: https://ownyourbits.com/2017/01/18/completely-remove-a-file-from-a-git-repository-with-git-forget-blob/

**Comment by mraible** [comment]
I haven’t done this yet, but hope to soon.

**Comment by AlainODea** [comment]
@mraible 👀 😁

**Comment by loopingz** [comment]
@AlainODea I think you can do it now as you are a collaborator :)

**Comment by AlainODea** [comment]
@loopingz I was thinking the same thing last night. I’ll take a crack at this soon. I’m on vacation and my Internet access is through my mobile phone, so I’m concerned about data charges or connection issues attempting that today.

**Comment by loopingz** [comment]
Enjoy your vacation, i would have helped but not the right status to do so :)

**Comment by mraible** [comment]
Done!

```
➜  okta-aws-cli-assume-role git:(master) git push --force origin master
Counting objects: 1306, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (421/421), done.
Writing objects: 100% (1306/1306), 249.93 KiB | 15.62 MiB/s, done.
Total 1306 (delta 481), reused 1306 (delta 481)
remote: Resolving deltas: 100% (481/481), done.
To ssh://github.com/oktadeveloper/okta-aws-cli-assume-role.git
 + 0810010...8849cdb master -> master (forced update)
Execution time: 9 s.
```

**Comment by AlainODea** [comment]
@mraible thank you!

There was one remaining snag keeping the repo and clones big: earlier release tags were holding references to the old history with the JARs.

I've corrected that and new clones are quite fast (under 2 seconds) and small (356.39 KiB transfer, 1.1MB including working copy) for me now.

Here are the changes I made to do that:
1. Move tag v1.0.0 from 844901bf626c161a737baeb0578e01548d6ff638 to a26c536efa17f4cf332d7679da0cbcf9a69b159e
2. Move tag v1.0.1 from 3b25ef644617857b36604118a444718d3807b7be to 7688a0ba6ab020c81da78ff9025099b439f64133
3. Move tag v1.0.2 from 3758c84bcb756bc3eaae98fd5bb473ae0a72bb27 to bfc37379001f9a82ff1adc466c898aeb289683d3
